veriscope_ta_dashboard/app/Rules/CryptoProofVerification.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+namespace App\Rules;
+
+use Illuminate\Contracts\Validation\Rule;
+use Illuminate\Support\Facades\Log;
+
+class CryptoProofVerification implements Rule
+{
+
+    public $expected_address;
+
+    public $trust_anchor_pubkey;
+
+    /**
+     * Create a new rule instance.
+     *
+     * @return void
+     */
+    public function __construct($expected_address, $trust_anchor_pubkey)
+    {
+      $this->expected_address = $expected_address;
+      $this->trust_anchor_pubkey = $trust_anchor_pubkey;
+    }
+
+    /**
+     * Replace single quotes with double quotes
+     *
+     * @return string
+     */
+    public function json_string_encode( $str ) {
+       // Replace the string passed
+       return str_replace('\'','"',$str);
+    }
+
+    /**
+     * Determine if the validation rule passes.
+     *
+     * @param  string  $attribute
+     * @param  mixed  $value
+     * @return bool
+     */
+    public function passes($attribute, $value)
+    {
+
+      // if crypto_proof is empty
+      if (empty($value)) {
+        return true;
+      }elseif (!file_exists("/opt/veriscope/veriscope_addressproof/test.py")) {
+        return true;
+      } else {
+        $escaped_data = base64_encode($this->json_string_encode($value));
```

Comment:
`passes` returns true (that the validation rules passed) if the addressproof files don't exist?